src/cascadia/TerminalApp/TerminalPage.h

```diff
@@ -6,12 +6,108 @@
 #include "winrt/Microsoft.UI.Xaml.Controls.h"
 
 #include "TerminalPage.g.h"
+#include "Tab.h"
+#include "CascadiaSettings.h"
+#include "Profile.h"
+
+#include <winrt/Microsoft.Terminal.TerminalControl.h>
+#include <winrt/Microsoft.Terminal.TerminalConnection.h>
+#include <winrt/Microsoft.UI.Xaml.Controls.Primitives.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
+#include <winrt/Microsoft.UI.Xaml.XamlTypeInfo.h>
 
 namespace winrt::TerminalApp::implementation
 {
     struct TerminalPage : TerminalPageT<TerminalPage>
     {
+    public:
         TerminalPage();
+
+        ~TerminalPage();
+
+        void SetSettings(::TerminalApp::CascadiaSettings* settings);
+
+        void Create();
+
+        hstring GetTitle();
+
+        void RefreshUIAfterSettingsReloaded();
+
+        void ShowOkDialog(const winrt::hstring& titleKey, const winrt::hstring& contentKey);
+
+        TerminalApp::TabRowControl* GetTabRow(); // This is a work-around because the winrt events are not working in Page
+
+        // -------------------------------- WinRT Events ---------------------------------
+        DECLARE_EVENT(TitleChanged, _titleChangeHandlers, winrt::Microsoft::Terminal::TerminalControl::TitleChangedEventArgs);
+        DECLARE_EVENT(LastTabClosed, _lastTabClosedHandlers, winrt::TerminalApp::LastTabClosedEventArgs);
+
+    private:
+        Microsoft::UI::Xaml::Controls::TabView _tabView{ nullptr };
+        TerminalApp::TabRowControl _tabRow{ nullptr };
+        Windows::UI::Xaml::Controls::Grid _tabContent{ nullptr };
+        Windows::UI::Xaml::Controls::SplitButton _newTabButton{ nullptr };
+
+        ::TerminalApp::CascadiaSettings* _settings;
+
+        std::vector<std::shared_ptr<Tab>> _tabs;
+
+        std::shared_mutex _dialogLock;
+
+        void _HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept;
+
+        void _CreateNewTabFromSettings(GUID profileGuid, winrt::Microsoft::Terminal::Settings::TerminalSettings settings);
+        void _OpenNewTab(std::optional<int> profileIndex);
+        void _CreateNewTabFlyout();
+
+        void _RegisterTerminalEvents(Microsoft::Terminal::TerminalControl::TermControl term, std::shared_ptr<Tab> hostingTab);
+
+        // Todo: add more event implementations here
+        // MSFT:20641986: Add keybindings for New Window
```

Comment:
Is there a better place for this comment?  This doesn't seem to go with _UpdateTitle or other immediate methods below.

Reply:
Moved to above "_scroll", also changed the order of the declarations of the methods